src/com/facebook/buck/jvm/groovy/GroovycStep.java

```diff
@@ -91,14 +98,61 @@ public String getDescription(ExecutionContext context) {
   }
 
   private ImmutableList<String> createCommand() {
-    ImmutableList.Builder<String> command = ImmutableList.builder();
-    command.addAll(groovyc.getCommandPrefix(resolver))
+    final ImmutableList.Builder<String> command = ImmutableList.builder();
+
+    command.addAll(groovyc.getCommandPrefix(resolver));
+
+    String classpath = Joiner.on(":").join(transform(declaredClasspathEntries, toStringFunction()));
+    command
         .add("-cp")
-        .add(Joiner.on(":").join(transform(declaredClasspathEntries, toStringFunction())))
+        .add(classpath.isEmpty() ? "''" : classpath)
         .add("-d")
-        .add(outputDirectory.toString())
-        .addAll(extraArguments.or(ImmutableList.<String>of()))
-        .addAll(transform(sourceFilePaths, toStringFunction()));
+        .add(outputDirectory.toString());
+    addCrossCompilationOptions(command);
+
+    command.addAll(extraArguments.or(ImmutableList.<String>of()))
+           .addAll(transform(sourceFilePaths, toStringFunction()));
     return command.build();
   }
+
+  private void addCrossCompilationOptions(final ImmutableList.Builder<String> command) {
+    if (shouldCrossCompile()) {
+      command.add("-j");
+      javacOptions.appendOptionsTo(new OptionsConsumer() {
+        @Override
+        public void addOptionValue(String option, String value) {
+          // javac won't find things compiled with groovyc otherwise.
+          // alternatively, we could convert all the paths to be absolute.
```

Comment:
So, by default in AbstractJavacOptions, we have this:

    // Set the sourcepath to stop us reading source files out of jars by mistake.
    optionsConsumer.addOptionValue("sourcepath", "");

It turns out that this doesn't play well with cross compilation, to the point where passing empty sourcepath to the 'inner' javac means it cannot find the groovyc compiled classes.

We found two potential paths to avoiding this problem; the simpler one (just ignore the source path option) is presented here. The more complicated solution involves making every path we pass to the compile step absolute; we liked this less.

I'm not particularly happy with either solution and would welcome any further input. I'll go and think about the possible values of source path to see if we always want to drop it, whether we want to log a warning if it's set, etc...